File: PrebidMobile/PrebidMobileRendering/Prebid/PBMCore/ORTB/Prebid/PBMORTBRewardedConfiguration.m
/*   Copyright 2018-2024 Prebid.org, Inc.

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

 http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */

#import "PBMORTBRewardedConfiguration.h"

#import "PBMORTBRewardedReward.h"
#import "PBMORTBRewardedCompletion.h"
#import "PBMORTBRewardedClose.h"

@implementation PBMORTBRewardedConfiguration

- (instancetype)initWithJsonDictionary:(PBMJsonDictionary *)jsonDictionary {
    if (!(self = [self init])) {
        return nil;
    }
    
    _reward = [[PBMORTBRewardedReward alloc] initWithJsonDictionary:[jsonDictionary[@"reward"] nullToNil]];
    _completion = [[PBMORTBRewardedCompletion alloc] initWithJsonDictionary:[jsonDictionary[@"completion"] nullToNil]];
    _close = [[PBMORTBRewardedClose alloc] initWithJsonDictionary:[jsonDictionary[@"close"] nullToNil]];
    
    return self;
}

- (PBMJsonDictionary *)toJsonDictionary {
    PBMMutableJsonDictionary * const ret = [[PBMMutableJsonDictionary alloc] init];
    
    ret[@"reward"] = [self.reward toJsonDictionary];
    ret[@"completion"] = [self.completion toJsonDictionary];
    ret[@"close"] = [self.close toJsonDictionary];
    
    [ret pbmRemoveEmptyVals];
    
    return ret;
}

@end

File: PrebidMobile/PrebidMobileRendering/Prebid/PBMCore/ORTB/Prebid/PBMORTBRewardedConfiguration.swift
//
// Copyright 2018-2025 Prebid.org, Inc.

// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at

// http://www.apache.org/licenses/LICENSE-2.0

// Unless required by applicable law or agreed to in writing, software
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
    

import Foundation

@objcMembers
public class PBMORTBRewardedConfiguration: PBMORTBAbstract {
    /// Metadata provided by the publisher to describe the reward.
    public var reward: PBMORTBRewardedReward?

    /// Describes the condition when the SDK should send a signal to the application that the user has earned the reward.
    public var completion: PBMORTBRewardedCompletion?

    /// Describes the close behavior. How should the SDK manage the ad when it is encountered as viewed.
    public var close: PBMORTBRewardedClose?
    
    private enum KeySet: String {
        case reward
        case completion
        case close
    }
    
    public override init() {
        super.init()
    }
    
    public override init(jsonDictionary: [String : Any]) {
        let json = JSONObject<KeySet>(jsonDictionary)
        
        reward = json[.reward]
        completion = json[.completion]
        close = json[.close]
        
        super.init()
    }
    
    public override func toJsonDictionary() -> [String : Any] {
        var json = JSONObject<KeySet>()
        
        json[.reward] = reward
        json[.completion] = completion
        json[.close] = close
        
        return json.dict
    }
}
